src/core/Object3D.js

```diff
@@ -835,5 +849,79 @@ Object3D.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 } );
 
+function Vector3WithDirtyFlag( x, y, z ) {
+
+	this._x = 0;
+	this._y = 0;
+	this._z = 0;
+
+	Vector3.call( this, x, y, z );
+
+	this._dirty = false;
```

Comment:
I don't wanna expose the existence of dirty flag to users then I added prefix `_` for `._dirty`.